include/capo/pcm.hpp

```diff
@@ -1,24 +1,31 @@
 #pragma once
 #include <capo/types.hpp>
+#include <capo/utils/format_size.hpp>
 #include <cstdint>
 #include <span>
 #include <string>
 #include <vector>
 
 namespace capo {
+enum class SampleFormat { eMono16, eStereo16 };
+
+struct SampleMeta {
+	std::size_t rate{};
+	SampleFormat format{};
+	std::uint8_t channels{};
+};
+
 ///
 /// \brief Uncompressed PCM data
 ///
 /// Only 16-bit Mono/Stereo is supported
 ///
 struct PCM {
-	enum class Format { eMono16, eStereo16 };
 	using Sample = std::int16_t;
 
+	SampleMeta meta;
 	std::vector<Sample> samples;
-	std::size_t sampleRate{};
-	Format sampleFormat{};
-	std::uint8_t channels{};
+	utils::Size size{};
```

Comment:
Why is `size` not in `meta`?

Reply:
Because it depends on the number of samples (aka is not metadata). Also, while that is currently just a single blob of all data, streaming may use a chunked/frames approach instead.